pkg/tfbridge/detailed_diff.go

```diff
@@ -153,6 +157,45 @@ func (differ detailedDiffer) getEffectiveType(path walk.SchemaPath) shim.ValueTy
 	return tfs.Type()
 }
 
+type (
+	setHash    int
+	arrayIndex int
+)
+
+type hashIndexMap map[setHash]arrayIndex
+
+func (differ detailedDiffer) calculateSetHashIndexMap(
+	path propertyPath, listVal []resource.PropertyValue,
+) hashIndexMap {
+	identities := make(hashIndexMap)
+
+	tfs, ps, err := lookupSchemas(path, differ.tfs, differ.ps)
+	if err != nil {
+		return nil
+	}
+
+	convertedVal, err := makeSingleTerraformInput(
+		differ.ctx, path.String(), resource.NewArrayProperty(listVal), tfs, ps)
+	if err != nil {
+		return nil
+	}
+
+	if convertedVal == nil {
+		return nil
+	}
+
+	convertedListVal, ok := convertedVal.([]interface{})
+	contract.Assertf(ok, "converted value should be a list")
+
+	// Calculate the identity of each element
+	for i, newElem := range convertedListVal {
+
+		elementHash := tfs.SetHash(newElem)
```

Comment:
You mentioned it can panic if we sent wrong data, might be reasonable to recover and bail.